hack/update-vendor-licenses.sh

```diff
@@ -196,6 +196,10 @@ for PACKAGE in $(go list -m -json all | jq -r .Path | sort -f); do
     echo "${PACKAGE} doesn't exist in ${DEPS_DIR}, skipping" >&2
     continue
   fi
+  if [[ -z "$(find "${DEPS_DIR}/${PACKAGE}" -mindepth 1 -maxdepth 1 \! -name 'v[0-9]*')" ]]; then
```

Comment:
This can match false-positive dirs and skip requiring licenses for code we build/ship. For example, the https://github.com/kubernetes/api/tree/master/core package has a directory named v1, but it is not a versioned submodule.

We should only exclude directories which 1) contain no files and 2) only have subfolders which directly contain a go.mod file (the v[0-9] pattern doesn't matter)

Reply:
thanks, that is a much better idea.  I've changed the script to implement it